src/source/Ice/IceAgent.c

```diff
@@ -2348,9 +2409,16 @@ STATUS handleStunPacket(PIceAgent pIceAgent, PBYTE pBuffer, UINT32 bufferLen, PS
                 pIceCandidatePair->roundTripTime = GETTIME() - requestSentTime;
                 DLOGD("Ice candidate pair %s_%s is connected. Round trip time: %" PRIu64 "ms", pIceCandidatePair->local->id,
                       pIceCandidatePair->remote->id, pIceCandidatePair->roundTripTime / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+                pIceCandidatePair->rtcIceCandidatePairDiagnostics.currentRoundTripTime = pIceCandidatePair->roundTripTime / HUNDREDS_OF_NANOS_IN_A_MILLISECOND;
+                pIceCandidatePair->rtcIceCandidatePairDiagnostics.totalRoundTripTime += pIceCandidatePair->roundTripTime / HUNDREDS_OF_NANOS_IN_A_MILLISECOND;
                 pIceCandidatePair->state = ICE_CANDIDATE_PAIR_STATE_SUCCEEDED;
             }
-
+            if (pIceCandidatePair == pIceAgent->pDataSendingIceCandidatePair && pIceCandidatePair != NULL) {
+                pIceAgent->pDataSendingIceCandidatePair->rtcIceCandidatePairDiagnostics.responsesReceived += connectivityCheckResponsesReceived;
+                pIceAgent->pDataSendingIceCandidatePair->rtcIceCandidatePairDiagnostics.lastResponseTimestamp = GETTIME();
+            }
+            pIceCandidatePair->rtcIceCandidatePairDiagnostics.responsesReceived += connectivityCheckResponsesReceived;
```

Comment:
are you double counting here when pIceCandidatePair is pDataSendingIceCandidatePair?

Reply:
My bad. It is redundant. I removed the updating of DataSendingCandidatePair because it would get updated in `iceAgentConnectedStateSetup` once state changes to `ICE_CANDIDATE_PAIR_STATE_SUCCEEDED`.